universal-login-contracts/test/contracts/GnosisSafe/deployProxy.ts

```diff
@@ -0,0 +1,56 @@
+import {expect} from 'chai';
+import {providers, Wallet, Contract, utils} from 'ethers';
+import {AddressZero} from 'ethers/constants';
+import {loadFixture, deployContract} from 'ethereum-waffle';
+import {createKeyPair, ETHER_NATIVE_TOKEN} from '@universal-login/commons';
+import {basicENS} from '@universal-login/commons/testutils';
+import {deployGnosisSafe, deployProxyFactory} from '../../../lib/gnosis-safe@1.1.1/deployContracts';
+import {encodeDataForSetup} from '../../../lib/gnosis-safe@1.1.1/encode';
+import {computeGnosisCounterfactualAddress} from '../../../lib/gnosis-safe@1.1.1/utils';
+import ENSUtils from '../../../build/TestableENSUtils.json';
+
+describe('GnosisSafe', async () => {
+  const domain = 'mylogin.eth';
+  const label = 'alex';
+  const hashLabel = utils.keccak256(utils.toUtf8Bytes(label));
+  const name = `${label}.${domain}`;
+  const node = utils.namehash(name);
+  let provider: providers.Provider;
+  let wallet: Wallet;
+  let gnosisSafe: Contract;
+  let proxyFactory: Contract;
+  let ensAddress: string;
+  let registrarAddress: string;
+  let publicResolver: string;
+  const keyPair = createKeyPair();
+
+  before(async () => {
+    ({wallet, publicResolver, registrarAddress, ensAddress, provider} = await loadFixture(basicENS));
+    gnosisSafe = await deployGnosisSafe(wallet);
+    proxyFactory = await deployProxyFactory(wallet);
+  });
+
+  it('deploys proxy and registers ENS name', async () => {
+    const ensRegistrar = await deployContract(wallet, ENSUtils as any);
+    const args = [hashLabel, name, node, ensAddress, registrarAddress, publicResolver];
+    const data = new utils.Interface(ENSUtils.interface as any).functions.registerENSUnderTests.encode(args);
```

Comment:
This is quite hackish, I created separate ticket to move ENS Registration to Gnosis Module.